src/rgw/rgw_dedup_utils.h

```diff
@@ -0,0 +1,442 @@
+#pragma once
+#include <string>
+#include "include/rados/buffer.h"
+#include "include/encoding.h"
+#include "common/Formatter.h"
+#include "common/ceph_json.h"
+#include <time.h>
+#include "include/utime.h"
+#include "include/encoding.h"
+
+//#define FULL_DEDUP_SUPPORT
+namespace rgw::dedup {
+  static constexpr const char* DEDUP_WATCH_OBJ = "DEDUP_WATCH_OBJ";
+  static constexpr uint64_t HEAD_OBJ_SIZE = 4*1024*1024; // 4MB
```

Comment:
this should be taken from the conf option, and also should not be defined in more than one place